DOCUMENTATION.md

```diff
@@ -0,0 +1,115 @@
+# Provider Documentation
+
+## Building and Reviewing
+
+The documentation is built from templates stored in the `templates`
+folder, along with examples stored in the `examples` folder. All of
+this content is copied to the `docs` folder, with the templates fully
+rendered, when the documentation is built.
+
+To validate the `templates` directory structure and its references to
+schemas in the provider and to examples, run:
+
+```shell
+make validate-docs
+```
+
+To build the documentation, replacing all of the contents of the
+`docs` folder:
+
+```shell
+make generate-docs
+```
+
+To preview a single documentation file, copy the content of the
+generated Markdown file into the [preview
+tool](https://registry.terraform.io/tools/doc-preview) on the
+HashiCorp website.
+
+## Templates
+
+Every resource and data source in the provider has its own
+documentation, which includes details of making use of that
+resource/data source, along with the details of its schema.
+
+If you do not write a template for a resource or data source that you
+create, `make generate-docs` will use a basic template when it
```

Comment:
"...you create, `make generate-docs` will use a basic template to generate the documentation..."
(Improves readability and avoids awkward repetition.)

Reply:
be7cebb2